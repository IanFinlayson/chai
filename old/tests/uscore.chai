def stuff(nums = [] [Int]):
    match nums:
        case [0, 0, 0]:
            print("All zeros")
        case [128, _, _]:
            print("128 in beginning!")
        case [_, 128, _]:
            print("128 in middle!")
        case [_, _, 128]:
            print("128 at end!")
        case ls:
            print("no 128...")

def main():
    stuff([0, 0, 0])
    stuff([8, 10, 128])
    stuff([43, 128, 555])
    stuff([128, -133, 3])
    stuff([131, -133, 3])

