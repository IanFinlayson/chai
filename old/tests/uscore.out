All zeros
128 at end!
128 in middle!
128 in beginning!
no 128...
